frontend/src/services/axiosBaseQuery.ts

```diff
@@ -0,0 +1,55 @@
+import { BaseQueryFn } from '@reduxjs/toolkit/query';
+import { AxiosRequestConfig, AxiosError } from 'axios';
+import axiosInstance from './instance'; // Import the instance with interceptors
+
+type AxiosBaseQueryArgs = {
```

Comment:
If it's possible, please use an existing Axios type instead of creating a new one. Maybe AxiosRequestConfig has the rest of the needed properties already.